### DISCUSSION TITLE
Delay syntax validation while caret moves.

### DISCUSSION THREAD

**Discussion by Lexyth** [godotengine/godot-proposals#11344] [Editor]
Currently, the syntax validation considers you to be done with what you were typing when you're not typing for a second.
This can lead to "work-in-progress" being marked as invalid before you're actually done with it, if, for instance, you're only moving the caret to another region of the word/statement.

As an example, changing any type to or from an Array version of that type marks the line as invalid while the caret is being moved in position to add/remove the closing bracket. 

Delaying the syntax validation until the user is done with typing AND moving the caret should more accurately reflect being done with the interaction as a whole.

Eg.: Delay validation by 1 second after you released any key, including arrow keys.

**Comment by Calinou** [comment]
This seems to be how VS Code works, so it makes sense to do the same in Godot:

https://github.com/user-attachments/assets/5196c308-2678-42fc-a98a-a99ab8f02da9

**Reply by Lexyth** [reply]
Yes, that is the behavior I would expect. 

On a side note, while "1 second after you released any key, including arrow keys" does describe an editing interaction ending, it does not cover all cases.

It would have to take the context the editing interaction has begun in into account and additionally consider the interaction complete if the caret moves outside of the highest relevant context. But that may be much more complicated and costly than it's worth.
On the other hand, faking this by considering the editing interaction done after a certain amount of time of only moving the caret without editing might be similar and simpler, but it does not accurately reflect the end of an editing interaction either and could cause premature validation.

Not that such an inaccuracy would really cause any problems, nor would addressing these really be noticeable. 
Take this more as a side note, just to have it mentioned.
